### PR TITLE
Tweak stylelint config

### PR DESCRIPTION
**Pull request by XhmikosR**
* move reportInvalidScopeDisables and reportNeedlessDisables to config
* rename to .stylelintrc.json
* inherit from the root config
* remove unneeded disabled from examples/.stylelintrc.json

### Type of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [ ] My change introduces changes to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

#### Live previews

- <https://deploy-preview-38228--twbs-bootstrap.netlify.app/>

### CODE REVIEW

**Comment by XhmikosR** [comment]
BTW I could use some help with replacing the deprecating rules in the new stylelint version https://github.com/twbs/stylelint-config-twbs-bootstrap/issues/188

File: site/content/docs/5.3/examples/.stylelintrc [hunk 1]
```diff
@@ -2,6 +2,8 @@
   "extends": [
     "stylelint-config-twbs-bootstrap"
```
Comment (XhmikosR) [inline site/content/docs/5.3/examples/.stylelintrc:3]: Might be better for maintenance if we inherited from the root .stylelintrc file. If anyone could push the change here, that'd help.

**Comment by XhmikosR** [comment]
I believe this is now ready. IMHO it's cleaner now.

The only issue I've been hitting is some caching issue when changing the config. But it's not something new IIRC.

**Review by mdo — APPROVED** [review]

**Comment by XhmikosR** [comment]
OK, now I really think it's ready :) Let me know which approach you prefer. I feel the latter is cleaner since those rules affect our Sass files only and we reduce the overrides.

**Review by mdo — APPROVED** [review]
Looks great!

**Pull request merged**